### ISSUE TITLE
Sync bookmarks and identities between devices

### ISSUE THREAD

**Issue by ghost** [skyjake/lagrange#517]
It would be great if bookmarks I make on my desktop were available on my mobile or other devices. 

For interactive Gemini capsules like Geddit, syncing identity might also be important so users can continue a conversation. 

Could be done via a 3rd party service API?

**Comment by skyjake** [comment]
I agree syncing would be great, but there are some non-technical hurdles here. The options are basically using some cloud storage API (like Dropbox, etc.) or putting together a custom sync server. Dealing with and adding dependencies on 3rd party APIs does not sound appealing, but neither does developing a reliable sync service (albeit a simple one). A custom server could be self-hosted, but that limits the appeal of the feature quite a bit as it removes a big part of the ease of use. I don't really want to run a sync service myself since that comes with certain reliability and uptime assumptions that I'm not in a position to meet at the moment.

As a manual solution, you can currently export and import data without overwriting old data, which solves part of the syncing problem.

**Comment by ghost** [comment]
They’re all fair points. Exporting sounds like the closest option for now so maybe I’ll go with that. 

If the user could change the default location of the bookmarks, maybe that would work? For example, if I set it to my Dropbox on my desktop and on my smartphone. Well, iCloud in my case.

**Comment by sjehuda** [comment]
Standard means can be utilized for that task.

---

I am a co-founder of project Blasta.

Blasta is an collaborative annotation management system (i.e. bookmarks manager) which purpose is to make bookmarks accessible from anywhere.

The experiment of Blasta has been successfull and I intend to write an extension for a Qt browser in order to integrate it gracefully into HTML browsers.

I advise to stongly consider to utilize the XMPP platform as a system to synchronize browser data, including bookmarks, history, tabs, cookies, password etc.

Chris:
> XMPP?

Schimon:
> Yes. XMPP, the *Extensible* Messaging and Presence Protocol.

Chris:
> This must be a mistake.

Schimon:
> **There is no mistake!**

Please. Watch the video presentation for the technicalities or visit these resources:
- https://blasta.woodpeckersnest.eu/help/about/xmpp/pubsub
- https://blasta.woodpeckersnest.eu/help/about/xmpp/atomsub

|          |                                                    |
|      --- | ---                                                |
| Code     | https://git.xmpp-it.net/sch/Blasta                 |
| Video    | https://video.xmpp-it.net/w/cfozoUeVLFbBFMCCSCJ1Dn |
| Instance | https://blasta.woodpeckersnest.eu                  |
|          |                                                    |

Resources:
- gemini://woodpeckersnest.space/~schapps/journal/2024-11-28-xmpp-for-browsers.gmi
- https://github.com/jarun/buku/discussions/795
